course4/homework1/vector2d.h

```diff
@@ -1,9 +1,71 @@
 #pragma once
 
+#include <math.h>
 #include <ostream>
+#include <stdexcept>
 
-struct Vector2d
+const double AngleToRadian = M_PI / 180;
+
+class Vector2d
 {
-	// TODO
+	double mX;
+	double mY;
+
+public:
+	Vector2d(double x, double y) : mX(x), mY(y) {}
+
+	double GetX() const { return mX;}
+	void SetX(double value) { mX = value; }
+
+	double GetY() const { return mY;}
+	void SetY(double value) { mY = value; }
+	
+	double GetRadius() const { return sqrt(mX*mX + mY*mY); }
+	double GetAngle() const
+	{
+		if (mX == 0)
+		{
+			if (mY == 0)
+				return 0; // not really defined
+
+			if (mY > 0)
+				return 90;
+
+			return 270;
+		}
+
+		if (mX < 0)
+			return atan(mY / mX) / AngleToRadian + 180;
+
+		if (mY < 0)
+			return atan(mY / mX) / AngleToRadian + 360;
+
+		return atan(mY / mX) / AngleToRadian;
```

Comment:
I would expect the function to return the angle in radians, like all the other math functions do